### PR TITLE
updated session docs

### PR DESCRIPTION
**Pull request by nelsonwittwer**
## Description

As noted in https://github.com/Shopify/shopify-api-ruby/issues/1163, our docs for sessions are out of date. These changes better outline how sessions work within the ruby libraries.

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -37,10 +37,25 @@ ShopifyAPI::Context.setup(
...
+
+Session persistence is handled by the [ShopifyApp](https://github.com/Shopify/shopify_app) gem and is recommended for use in the Rails context. See that gem for documentation on how to use it.
+
+#### Cookie
+Cookie based authentication is not supported for embedded apps due to browsers dropping support for third party cookies due to security concerns. Non-embedded apps are able to use cookies for session storage/retrieval.
+
+For *non-embedded* apps, you can pass the cookies into `ShopifyAPI::Utils::SessionUtils.current_session_id(nil, cookies, true)` for online (user) sessions or `ShopifyAPI::Utils::SessionUtils.current_session_id(nil, cookies, false)` for offline (store) sessions.
+
+#### Getting Session ID From Embedded Requests
+For *embedded* apps, you can pass the auth header into `ShopifyAPI::Utils::SessionUtils.current_session_id(auth_header, nil, true)` for online (user) sessions or `ShopifyAPI::Utils::SessionUtils.current_session_id(auth_header, nil, false)` for offline (store) sessions.
```
Comment (paulomarg) [inline docs/getting_started.md:52]: Is it worth mentioning how to get the bearer token (e.g. calling `authenticatedFetch` from app bridge) here?

**Review by paulomarg — APPROVED** [review]
Just a nit!

**Comment by vfonic** [comment]
Where in the docs I can read more about `auth_header` variable that appears in several places in the docs out of nowhere?

**Comment by kyreeves** [comment]
@vfonic I'm having trouble making requests in background jobs as well. Have you had any luck with this since your last comment?

**Comment by vfonic** [comment]
Yes, I have it fully working. You can do:

```ruby
shop.with_shopify_session do
  # any code that makes requests to Shopify REST or GraphQL API
end
```

or

```ruby
session = ShopifyAPI::Auth::Session.new(shop: shop.shopify_domain, access_token: shop.shopify_token, scope: shop.access_scopes)
# and then
client = ShopifyAPI::Clients::Graphql::Admin.new(session: session)
client.query(query:, variables:).body
# or
ShopifyAPI::Context.activate_session(session)
ShopifyAPI::Webhook.all.as_json
# or whatever you need
```

Something along these lines should work. In both cases `shop` is an ActiveRecord instance of `Shop` model as per ShopifyApp gem.

**Pull request merged**